Sources/LiveViewNativeAVKit/VideoPlayerView.swift

```diff
@@ -0,0 +1,232 @@
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    var autoplay: Bool = false
+
+    /// The URL of the video to play.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    var url: URL
+
+    @LiveContext<R> private var context
+    @ObservedElement private var element: ElementNode
+    @StateObject private var observer: VideoPlayerObserver<R>
+
+    init(element: ElementNode) {
+        if let value = element.attributeValue(for: "autoplay") {
+            self.autoplay = Bool(value)!
+        }
+        if let value = element.attributeValue(for: "url") {
+            self.url = URL(string: value)!
+        } else {
+            self.url = URL(string: "")!
+        }
+        let interval = CMTime(seconds: 0.01, preferredTimescale: CMTimeScale(NSEC_PER_SEC))
+
+        _observer = StateObject(wrappedValue: VideoPlayerObserver(interval: interval))
+    }
+
+    var body: some View {
+        VideoPlayer(player: self.observer.player)
+            .onAppear {
+                timeControlStatus = "paused"
+                observer.player.replaceCurrentItem(with: AVPlayerItem(url: url))
+
+                if self.autoplay {
+                    self.observer.player.play()
+                }
+            }
+            .onReceive(context.coordinator.receiveEvent("pause"), perform: performPause)
+            .onReceive(context.coordinator.receiveEvent("play"), perform: performPlay)
+            .onReceive(context.coordinator.receiveEvent("seek"), perform: performSeek)
+            .onReceive(observer.player.publisher(for: \.isMuted)) { value in isMuted = value }
+            .onReceive(observer.player.publisher(for: \.timeControlStatus)) { value in syncTimeControlStatus(status: value) }
+            /// This is a workaround for SwiftUI not supporting `onReceive` for `AVPlayer`'s `currentTime` property.
+            .onReceive(observer.assigns.$playbackTime.throttle(for: AVKitHelpers.floatToStride(float: playbackTimeUpdateInterval), scheduler: RunLoop.current, latest: true)) { value in playbackTime = value }
+    }
+
+    func performPlay(params: Dictionary<String, Any>) {
+        observer.player.play()
+    }
+
+    func performPause(params: Dictionary<String, Any>) {
+        observer.player.pause()
+    }
+
+    func performSeek(params: Dictionary<String, Any>) {
+        switch (params["playback_time"]) {
+        case .some(let value):
+            let castValue = value as! Float64
+            let to = AVKitHelpers.floatToCMTime(float: castValue)
+            let toleranceBefore = AVKitHelpers.floatToCMTime(float: 0.0)
+            let toleranceAfter = AVKitHelpers.floatToCMTime(float: 0.0)
+
+            observer.player.seek(to: to, toleranceBefore: toleranceBefore, toleranceAfter: toleranceAfter)
+
+            // Always update playback time when seeking.
+            playbackTime = CMTimeGetSeconds(observer.player.currentTime())
+
+        case .none:
+            return
+        }
```

Comment:
`guard let` is a bit more clear than a switch, and avoids a force cast.
```suggestion
        guard let value = params["playback_time"] as? Float64 else { return }
        
        let to = AVKitHelpers.floatToCMTime(float: value)
        let toleranceBefore = AVKitHelpers.floatToCMTime(float: 0.0)
        let toleranceAfter = AVKitHelpers.floatToCMTime(float: 0.0)

        observer.player.seek(to: to, toleranceBefore: toleranceBefore, toleranceAfter: toleranceAfter)

        // Always update playback time when seeking.
        playbackTime = CMTimeGetSeconds(observer.player.currentTime())
```